server.js

```diff
@@ -0,0 +1,118 @@
+var express = require('express');
+var request = require('request'); // "Request" library
+var querystring = require('querystring');
+var cookieParser = require('cookie-parser');
+var port = process.env.PORT || 8888;
+
+var client_id = process.env.clientId; // Your client id
+var client_secret = process.env.clientSecret; // Your client secret
+var redirect_uri = 'http://spotavinyl.herokuapp.com/callback'; // Your redirect uri
+
+/**
+ * Generates a random string containing numbers and letters
+ * @param  {number} length The length of the string
+ * @return {string} The generated string
+ */
+var generateRandomString = function(length) {
+  var text = '';
+  var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+
+  for (var i = 0; i < length; i++) {
+    text += possible.charAt(Math.floor(Math.random() * possible.length));
+  }
+  return text;
+};
+
+var stateKey = 'spotify_auth_state';
+
+var app = express();
```

Comment:
Typically declare these vars up at the top of the file. That way we know what `app` is.